### PR TITLE
Draft chapter about documentation with man

### LINKED ISSUE #70: Book: Man page generation draft

### PR DESCRIPTION
**Pull request by spacekookie**
Fixes #70

### CODE REVIEW

File: src/in-depth/docs.md [hunk 1]
```diff
@@ -1 +1,64 @@
...
+
+Documentation for CLIs usually consists of
+a `--help` section in the command
+and a manual (`man`) page.
+
+Both can be automatically generated
+when using `clap`,
+via the `man` backend.
+
+Let's assume you use `structopt`
```
Comment (killercup) [inline src/in-depth/docs.md]: you say structopt but your code say `#[derive(Clap)]` ;)

File: src/in-depth/docs.md [hunk 2]
```diff
@@ -1 +1,64 @@
...
+to create a CLI.
+
+```rust
+/// It's like `cat` but in Rust
```
Comment (killercup) [inline src/in-depth/docs.md]: wait it's actually like head, though? maybe call it `front`, …or `skull`

File: src/in-depth/docs.md [hunk 3]
```diff
@@ -1 +1,64 @@
...
+    #[clap(parse(from_os_str))]
+    pub file: PathBuf,
+    /// how many lines to print
+    #[clap(short = "n", default_value = "5")]
+    pub count: usize,
+}
+```
+
+Secondly, you need to use a `build.rs`
+to generate a manual at compile-time
```
Comment (killercup) [inline src/in-depth/docs.md]: > a manual

maybe "the manual file"

File: src/in-depth/docs.md [hunk 4]
```diff
@@ -1 +1,64 @@
...
+from the definition of your app
+in code.
+
+There are a few things to keep in mind
+such as how you want to package your binary
```
Comment (killercup) [inline src/in-depth/docs.md]: maybe this part of the sentence in parentheses?

File: src/in-depth/docs.md [hunk 5]
```diff
@@ -1 +1,64 @@
...
+but for now
+we simply put the `man` page
```
Comment (killercup) [inline src/in-depth/docs.md]: page -> file?

File: src/in-depth/docs.md [hunk 6]
```diff
@@ -1 +1,64 @@
...
+```rust
+extern crate clap;
+extern crate clap_generate;
+
+#[path="src/cli.rs"]
+mod cli;
+
+fn main() {
+    use clap::IntoApp;
+    let app = cli::Head::into_app();
```
Comment (killercup) [inline src/in-depth/docs.md]: "Head" <- busted!

File: src/in-depth/docs.md [hunk 7]
```diff
@@ -1 +1,64 @@
...
+
+    use clap_generate::gen_manuals;
+    for man in gen_manuals(&app) {
+        let name = "rat.1"; // FIXME: Extract this from man!
```
Comment (killercup) [inline src/in-depth/docs.md]: eh, maybe omit the fixme

File: src/in-depth/docs.md [hunk 8]
```diff
@@ -1 +1,64 @@
...
+        let mut out = fs::File::create("rat.1").unwrap();
+        use std::io::Write;
+        out.write_all(man.render().as_bytes()).unwrap();
+    }
+}
+```
+
+When you now compile your application
+there will be a `rat.1` file 
+in your project director.
```
Comment (killercup) [inline src/in-depth/docs.md]: director[y]

File: src/in-depth/docs.md [hunk 9]
```diff
@@ -1 +1,64 @@
...
+
+If you open that in `man`
```
Comment (killercup) [inline src/in-depth/docs.md]: how do you do that?

File: src/in-depth/docs.md (hunk 6 above)
Comment (spacekookie) [reply src/in-depth/docs.md]: :scream:

Comment (spacekookie) [reply src/in-depth/docs.md]: I tried to be clever, okay? :joy:

File: src/in-depth/docs.md (hunk 7 above)
Comment (spacekookie) [reply src/in-depth/docs.md]: That was more meant as an FYI to anyone editing the book :stuck_out_tongue_winking_eye:

File: src/in-depth/docs.md (hunk 9 above)
Comment (spacekookie) [reply src/in-depth/docs.md]: `woman ./rat.1`

File: src/in-depth/docs.md [hunk 10]
```diff
@@ -1 +1,61 @@
...
+Documentation for CLIs usually consists of
+a `--help` section in the command
+and a manual (`man`) page.
+
+Both can be automatically generated
+when using `clap`,
+via the `man` backend.
+
+```rust
+/// It's like `cat` but in Rust
```
Comment (killercup) [inline src/in-depth/docs.md]: 🐈?

File: src/in-depth/docs.md [hunk 11]
```diff
@@ -1 +1,61 @@
...
+#[path="src/cli.rs"]
+mod cli;
+
+fn main() {
+    use clap::IntoApp;
+    let app = cli::Head::into_app();
+
+    use clap_generate::gen_manuals;
+    for man in gen_manuals(&app) {
+        let name = "rat.1";
```
Comment (killercup) [inline src/in-depth/docs.md]: 🐀?

**Pull request merged**